contracts/Migrator.sol

```diff
@@ -87,112 +108,73 @@ contract Migrator is Governable {
-            // TODO: Do we need some sort of tolerance (may be 0.01%)?
-            revert BalanceMismatch(netOgvAmountIn, netExpectedOgnAmountOut, netOgnAmountOut);
-        }
+    function isMigrationActive() public view returns (bool) {
+        return endTime > 0 && block.timestamp < endTime;
     }
 
     /**
      * @notice Migrates the specified amount of OGV to OGN
      * @param ogvAmount Amount of OGV to migrate
      * @return ognReceived OGN Received
      */
-    function migrate(uint256 ogvAmount) external netBalanceCheck returns (uint256 ognReceived) {
+    function migrate(uint256 ogvAmount) external isSolvent returns (uint256 ognReceived) {
         return _migrate(ogvAmount, msg.sender);
     }
 
-    /**
-     * @notice Migrates all of user's OGV to OGN
-     * @return ognReceived OGN Received
-     */
-    function migrateAll() external netBalanceCheck returns (uint256 ognReceived) {
-        return _migrate(ogv.balanceOf(msg.sender), msg.sender);
-    }
-
-    /**
-     * @notice Migrates caller's OGV to OGN and sends it to the `receiver`
-     * @return ognReceived OGN Received
-     */
-    function _migrate(uint256 ogvAmount, address receiver) internal returns (uint256 ognReceived) {
-        if (!isMigrationActive()) {
-            revert MigrationIsInactive();
-        }
-
-        ognReceived = ogvAmount * CONVERSION_RATE / 1 ether;
-
-        emit TokenExchanged(ogvAmount, ognReceived);
-
-        ogv.transferFrom(msg.sender, address(this), ogvAmount);
-
-        if (receiver != address(this)) {
-            // When migrating stakes, the contract would directly
-            // stake the balance on behalf of the user. So there's
-            // no need to transfer to self. Transfering to user and then
-            // back to this contract would only increase gas cost (and
-            // an additional tx for the user).
-            ogn.transfer(receiver, ognReceived);
-        }
-    }
-
     /**
      * @notice Migrates OGV stakes to OGN. Can also include unstaked OGN & OGV
      *          balances from the user's wallet (if specified).
      * @param lockupIds OGV Lockup IDs to be migrated
      * @param ogvAmountFromWallet Extra OGV balance from user's wallet to migrate & stake
      * @param ognAmountFromWallet Extra OGN balance from user's wallet to stake
      * @param migrateRewards If true, Migrate & Stake received rewards
+     * @param newStakeAmount Max amount of OGN (from wallet+unstake) to stake
      * @param newStakeDuration Duration of the new stake
      */
     function migrate(
         uint256[] calldata lockupIds,
         uint256 ogvAmountFromWallet,
         uint256 ognAmountFromWallet,
         bool migrateRewards,
+        uint256 newStakeAmount,
         uint256 newStakeDuration
-    ) external netBalanceCheck {
+    ) external isSolvent {
         if (!isMigrationActive()) {
             revert MigrationIsInactive();
         }
 
+        if (newStakeAmount == 0) {
```

Comment:
It's fine for new stake amount to be zero - we just don't do a stake in that case.